src/components/DonationItemsTable/DonationsModal.jsx

```diff
@@ -0,0 +1,117 @@
+import { useEffect, useState } from 'react';
+import { useBackend } from '../../contexts/BackendContext';
+import PropTypes from 'prop-types';
+import { Input, Text, Modal, ModalContent, ModalHeader, ModalBody, ModalOverlay, ModalFooter, IconButton, Button} from '@chakra-ui/react';
+import {CloseIcon} from '@chakra-ui/icons';
+
+const DonationsModal = ({ isOpen, onClose, data, setData, currentPageNum }) => {
+  const { backend } = useBackend();
+  const [categoryData, setCategoryData] = useState('');
+  const [typeData, setTypeData] = useState('');
+  const [weightData, setWeightData] = useState('');
+  const [priceData, setPriceData] = useState('');
+
+  useEffect(() => {
+    const getData = async () => {
+      try {
+        if (isOpen && data) {
+          setCategoryData(data['category']);
+          setTypeData(data['item_name']);
+          setWeightData(data['quantity_type']);
+          setPriceData(data['price']);
+        }
```

Comment:
make sure that the data present isn't the data that was just passed to the post request. otherwise we get a little bug where when you create an item and then click add item, it populates the modal with the data from the last item you made.

Reply:
maybe when you call the post request what you can do is call all the set useStates to make the data empty